faunadb-java/src/main/java/com/faunadb/client/query/Expr.java

```diff
@@ -0,0 +1,230 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Ref;
+import com.faunadb.client.types.SetRef;
+import com.faunadb.client.types.Value;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * A query language expression. Constructors for this class are at the {@link Language} class.
+ */
+public abstract class Expr implements Value {
```

Comment:
I think this is the wrong subtype relationship. Value should be a subtype of Expr, so that a Value can be passed directly as an Expr (call-sites that take an Expr will have to do a run-time check to determine if escaping needs to happen). Call-sites that expect a Value do not have to handle the case when an Expr is passed.